src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java

```diff
@@ -293,11 +293,11 @@ public boolean equals(Object object) {
             return false;
         }
         final LocalizedMessage localizedMessage = (LocalizedMessage) object;
-        return Objects.equals(lineNo, localizedMessage.lineNo)
+        return severityLevel == localizedMessage.severityLevel
+                && Objects.equals(lineNo, localizedMessage.lineNo)
```

Comment:
This change required by TC now produces this violation:
````
LocalizedMessage.java (1)
296: equals() Overly complex boolean expression (11 terms) 
````

Reply:
@rnveach , we did interesting update:
to fix style issue, we make code is not consistent in view so break style consistency.
BUT it and this is still forcing us to suppress smth else.
As original violation is not performance issue fix .... damaging existing style might not bee good. 

i think it actually suppress original issue and keep code consistency.

Reply:
I agree. I will suppress it.